src/Agent.Worker/JobRunner.cs

```diff
@@ -279,6 +279,27 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                     }
                 }
 
+#if OS_WINDOWS
+            bool retainDefaultEncoding = jobContext.Variables.GetBoolean(Constants.Variables.Agent.RetainDefaultEncoding) ?? false;
```

Comment:
phase 1: we release an agent version that is able to run `chcp`, but will not run it by default.
phase 2: we turn on a feature flag on server side Ring by Ring, server will set `RetainDefaultEncoding=false`, so agent start running `chcp`
phase 3: we release another agent version that will run `chcp` by default, we will remove the feature flag on server. customer who is having issue can manually set `RetainDefaultEncoding=true`

Reply:
i think we are in phase 1, so `?? true` would be better.

Reply:
That makes sense, I was thinking we'd basically just do it slightly differently:
phase 1: Turn on a feature flag that sets ```retainDefaultEncoding = true```
phase 2: Merge this PR
phase 3: Turn off feature flag (probably leave the logic though so it can quickly be turned back on if needed)
phase 4: Remove the feature flag entirely

I think the (admittedly minor) advantage here is that all of the turning on/off logic is controlled by the feature flag and we don't have to touch the agent again. The only reason I'd been getting this PR ready first was so that we could make sure it works before worrying about introducing the feature flag.

Reply:
I also updated accordingly for the rest of the recommendations you made!

Reply:
Update, talked with Ting offline and we're going to go with his original suggestion. This allows us to release the agent even if the feature flag isn't ready which is beneficial since the agent can be released more quickly (especially if it misses next week's sprint cutoff in VSO)

Reply:
Update, brought Eric in to the conversation, decided to go with my original way to make sure that only one version of this exists on agents (even when some agents haven't been updated)